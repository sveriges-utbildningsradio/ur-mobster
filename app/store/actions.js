import * as types from './actionTypes'

export const updateLanguage = language => ({
  type: types.UPDATE_LANGUAGE,
  payload: language
})

export const updateDuration = duration => ({
  type: types.UPDATE_DURATION,
  payload: duration
})

export const updateBreakDuration = duration => ({
  type: types.UPDATE_BREAK_DURATION,
  payload: duration
})

export const updateBreakFrequency = frequency => ({
  type: types.UPDATE_BREAK_FREQUENCY,
  payload: frequency
})
