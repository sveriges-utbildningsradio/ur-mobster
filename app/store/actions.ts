import * as types from './actionTypes'
import { LanguageValue } from '../types'

export const updateLanguage = (language: LanguageValue) => ({
  type: types.UPDATE_LANGUAGE,
  payload: language
})

export const updateDuration = (duration: number) => ({
  type: types.UPDATE_DURATION,
  payload: duration
})

export const updateBreakDuration = (duration: number) => ({
  type: types.UPDATE_BREAK_DURATION,
  payload: duration
})

export const updateBreakFrequency = (frequency: number) => ({
  type: types.UPDATE_BREAK_FREQUENCY,
  payload: frequency
})

export const updateBreakTimeLeft = (timeLeft: number) => ({
  type: types.UPDATE_BREAK_TIME_LEFT,
  payload: timeLeft
})
