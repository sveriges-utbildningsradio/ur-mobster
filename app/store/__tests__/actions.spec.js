import * as actions from '../actions'

describe('store/actions', () => {
  it('updateLanguage', () => {
    const action = actions.updateLanguage('en')

    expect(action).toMatchSnapshot()
  })

  it('updateDuration', () => {
    const action = actions.updateDuration(300)

    expect(action).toMatchSnapshot()
  })

  it('updateBreakDuration', () => {
    const action = actions.updateBreakDuration(300)

    expect(action).toMatchSnapshot()
  })

  it('updateBreakFrequency', () => {
    const action = actions.updateBreakFrequency(300)

    expect(action).toMatchSnapshot()
  })

  it('updateBreakTimeLeft', () => {
    const action = actions.updateBreakTimeLeft(300)

    expect(action).toMatchSnapshot()
  })
  it('updateTimeLeft', () => {
    const action = actions.updateTimeLeft(300)

    expect(action).toMatchSnapshot()
  })

  it('updateIsRunning', () => {
    const action = actions.updateIsRunning(true)

    expect(action).toMatchSnapshot()
  })
  it('updateReachedEnd', () => {
    const action = actions.updateReachedEnd(true)

    expect(action).toMatchSnapshot()
  })

  it('updateTimeSinceBreak', () => {
    const action = actions.updateTimeSinceBreak(300)

    expect(action).toMatchSnapshot()
  })

  it('updateIsOnBreak', () => {
    const action = actions.updateIsOnBreak(true)

    expect(action).toMatchSnapshot()
  })
})
