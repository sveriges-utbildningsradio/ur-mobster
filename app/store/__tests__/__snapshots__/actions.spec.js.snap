// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/actions updateBreakDuration 1`] = `
Object {
  "payload": 300,
  "type": "UPDATE_BREAK_DURATION",
}
`;

exports[`store/actions updateBreakFrequency 1`] = `
Object {
  "payload": 300,
  "type": "UPDATE_BREAK_FREQUENCY",
}
`;

exports[`store/actions updateBreakTimeLeft 1`] = `
Object {
  "payload": 300,
  "type": "UPDATE_BREAK_TIME_LEFT",
}
`;

exports[`store/actions updateDuration 1`] = `
Object {
  "payload": 300,
  "type": "UPDATE_DURATION",
}
`;

exports[`store/actions updateLanguage 1`] = `
Object {
  "payload": "en",
  "type": "UPDATE_LANGUAGE",
}
`;
